src/UI/Components/Diagram/helpers.ts

```diff
@@ -0,0 +1,38 @@
+import { ServiceInstanceModel, ServiceModel } from "@/Core";
+
+export const extractRelationsIds = (
```

Comment:
Same observation as for the other method regarding the different attribute state types, would be nice to later on to remove the repetition.